docs/learn/encyclopedia/data-format/xdr.mdx

```diff
@@ -38,3 +41,219 @@ XDR is specified in [RFC 4506](http://tools.ietf.org/html/rfc4506.html) and is s
 Since XDR is a binary format and not as widely known as simpler formats like JSON, the Stellar SDKs all include tools for parsing XDR and will do so automatically when retrieving data.
 
 In addition, the Horizon API server generally exposes the most important parts of the XDR data in JSON, so they are easier to parse if you are not using an SDK. The XDR data is still included (encoded as a base64 string) inside the JSON in case you need direct access to it.
+
+## Digging into XDR structures
+
+Since the XDR format is a fundamental underpinning of the Stellar Network, we often find it necessary to dig into these raw binary data structures for clients to interact with things like transactions and authorization entries. It can be hard to understand because of its structure, but this section will aim to enlighten you on the ways to interact with it in a handful of popular languages. The Protocol's schema is defined in the [`stellar/stellar-xdr`](https://github.com/stellar/stellar-xdr/) repository, though that contains far more than what we'll need for our purposes.
+
+### XDR's Common Forms
+
+In the Stellar Protocol, XDR takes a handful of different forms that are important to understand and distinguish:
+
+#### Basic Primitives
+
+These are the easiest to understand for anyone familiar with a programming language and includes things like integers, strings, and arrays of bytes. These are fairly straightforward to interact with, though things can get more complicated when you have aliases.
+
+For example, you have the `Uint64` alias in the Stellar Protocol (defined [Stellar-types.x](https://github.com/stellar/stellar-xdr/blob/70180d5e8d9caee9e8645ed8a38c36a8cf403cd9/Stellar-types.x#L14)), defined for readability instead of XDR's native "unsigned hyper" value. You can treat it exactly as you'd expect:
+
+<CodeExample>
+
+```typescript
+import { xdr } from "@stellar/stellar-sdk";
+
+const u64 = new xdr.Uint64(12345678);
+```
+
+</CodeExample>
```

Comment:
````suggestion
```typescript
import { xdr } from "@stellar/stellar-sdk";

const u64 = new xdr.Uint64(12345678);
```
````

The `<CodeExample>` component is really only useful when you're combining two (or more) snippets from different languages, illustrating the same thing (as you've done some places in this doc). When you're just using JS like you are here, it's fine to omit the code example component.